builtin/providers/aws/resource_aws_glacier_vault.go

```diff
@@ -163,8 +163,14 @@ func resourceAwsGlacierVaultRead(d *schema.ResourceData, meta interface{}) error
 	if awserr, ok := err.(awserr.Error); ok && awserr.Code() == "ResourceNotFoundException" {
 		d.Set("access_policy", "")
 	} else if pol != nil {
-		policy, _ := normalizeJsonString(*pol.Policy.Policy)
-		d.Set("access_policy", policy)
+		policy, err := normalizeJsonString(*pol.Policy.Policy)
+		if err != nil {
+			return fmt.Errorf("access policy contains an invalid JSON: %s", policy)
```

Comment:
Is this the new and accepted way? I have seen a lot of request without this. It seem it depends on the point of view and/or approach the contributor has, which is not very consistent.

I will fix this.